miner/bid_simulator.go

```diff
@@ -442,10 +440,19 @@ func (b *bidSimulator) clearLoop() {
 func (b *bidSimulator) sendBid(_ context.Context, bid *types.Bid) error {
 	timer := time.NewTimer(1 * time.Second)
 	defer timer.Stop()
+
+	replyCh := make(chan error)
+
 	select {
-	case b.newBidCh <- bid:
+	case b.newBidCh <- newBidPackage{bid: bid, feedback: replyCh}:
 		b.AddPending(bid.BlockNumber, bid.Builder, bid.Hash())
-		return nil
+	case <-timer.C:
+		return types.ErrMevBusy
+	}
+
+	select {
+	case reply := <-replyCh:
+		return reply
```

Comment:
because the `replyCh` get no buffer, if we return here directly, it is possible another routinue will be blocked when sending item into `replyCh` and thus cause dead lock.